src/utils.ts

```diff
@@ -1,6 +1,28 @@
 import { Result, ok, err } from './result'
 import { ResultAsync } from './result-async'
 
+/**
+ * Wraps a function with a try catch, creating a new function with the same
+ * arguments but returning `Ok` if successful, `Err` if the function throws
+ *
+ * @param fn function to wrap with ok on success or err on failure
+ * @param errorFn when an error is thrown, this will wrap the error result if provided
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function fromThrowable<Fn extends (...args: readonly any[]) => any, E>(
```

Comment:
I'm fairly confident that you can replace usage of `any` by using the `infer` keyword in this case. Give it a shot.

```suggestion
export function fromThrowable<Fn extends (...args: readonly (infer U)[]) => any, E>(
```

More info:

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html#type-inference-in-conditional-types

---

Secondly, you could get rid of the `any` return type by declaring a additional type argument.

```suggestion
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function fromThrowable<T, E, Fn extends (...args: readonly (infer U)[]) => T>(
```

Reply:
Unfortunately trying `infer U` causes a typescript error
> `'infer' declarations are only permitted in the 'extends' clause of a conditional type.`

This can be changed to `unknown` with no adverse effect
```suggestion
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function fromThrowable<Fn extends (...args: readonly unknown[]) => any, E>(
```
---

Changing the return type by declaring an additional type argument T, makes the response OK be `unknown`

`() => Result<unknown, E>`.

Reply:
The `any` is not leaked to the return type so I think we can go ahead with this? 
I happy to remove the `any`, but am unsure of an approach at the moment